### PR TITLE
feat: fetch the number of stars dynamically

### PR DESCRIPTION
**Pull request by orhun**
This PR simply uses the GitHub API to fetch the number of stars instead of using a hardcoded value. We can potentially do the same thing for other values (such as contributors) - just let me know if this is applicable.

Screenshot 2023-10-29 at 11-27-40 Shuttle

A couple of points:

- We need to add a GitHub API token to the environment for this to work without aggressive GitHub rate limiting - just create a token without any permissions and set it in the environment as `GITHUB_ACCESS_TOKEN`.
- We should consider if we want to show "Loading" or an arbitrary number instead (e.g. `5k+`)

### CODE REVIEW

File: components/sections/CommunitySupportedNumbers.tsx [hunk 1]
```diff
@@ -1,9 +1,30 @@
...
+import { useEffect, useState } from 'react'
 
 const CommunitySupportedNumbers = () => {
+	const [stars, setStars] = useState(null);
+	const githubToken = process.env.GITHUB_ACCESS_TOKEN;
+	useEffect(() => {
+		const apiUrl = `https://api.github.com/repos/${GITHUB_REPO}`;
+		const options = {
+			headers: {
+				Authorization: `token ${githubToken}`,
```
Comment (GlenDC) [inline components/sections/CommunitySupportedNumbers.tsx:13]: FYI, for <https://plabayo.tech/> I get the star count service from the same API. As long as you cache it long enough and given that you only use this one endpoint, you should be fine. You can do up to 60 requests / hour without authentication.

Just saying, as the less API keys you need to keep around and protect, the more secure your posture is by definition. At least that's what i think about it .

Comment (GlenDC) [reply components/sections/CommunitySupportedNumbers.tsx:13]: And I have chosen to preload it at startup myself and only refresh it on success. This way it’s non blocking and always available

**Comment by jonaro00** [comment]
Great suggestions! @GlenDC

File: components/sections/CommunitySupportedNumbers.tsx (hunk 1 above)
Comment (orhun) [reply components/sections/CommunitySupportedNumbers.tsx:13]: I don't know how to cache it, do you know a good way of doing that?

**Pull request merged**